app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rposts/RPostViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.sharewhatyoucanproject.rposts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.GeoPoint
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlin.collections.ArrayList
+
+class RPostViewModel(
```

Comment:
Could you use more descriptive names for these classes?

Maybe:

```
PostDetailsViewModel
PostDetailsFragment
```
and

```
PostListViewModel
PostListFragment
```

Reply:
I actually thought it would be easier to differentiate receiver and donor view posts for receiver and view listings for donor.  

-Then I will name this way:

PostListViewModel.        PostDetailsViewModel         
PostListFragment.           PostDetailsFragment
for Receiver.


And ViewListingsFragment
        ViewListingsViewModel

Reply:
If the data model is the same you could use the same view model, just modify the method so that it will fetch either nearby posts or a donor’s posts they’ve listed

Reply:
Okay Morgan. But actually I'm using different data model. I will change it.